src/plot_api/plot_api.js

```diff
@@ -2508,6 +2527,360 @@ Plotly.relayout = function relayout(gd, astr, val) {
+        // time. In other words, if there's an axis transition, then set the data transition
+        // to instantaneous.
+        if(hasAxisTransition) {
+            traceTransitionConfig = Lib.extendFlat({}, transitionConfig);
+            traceTransitionConfig.duration = 0;
+        } else {
+            traceTransitionConfig = transitionConfig;
+        }
+
+        for(j = 0; j < basePlotModules.length; j++) {
+            // Note that we pass a callback to *create* the callback that must be invoked on completion.
+            // This is since not all traces know about transitions, so it greatly simplifies matters if
+            // the trace is responsible for creating a callback, if needed, and then executing it when
+            // the time is right.
+            basePlotModules[j].plot(gd, transitionedTraces, traceTransitionConfig, makeCallback);
+        }
+
+        // If nothing else creates a callback, then this will trigger the completion in the next tick:
+        setTimeout(makeCallback());
+
+        if(!hasAxisTransition && !hasTraceTransition) {
+            return false;
+        }
+
+    }
+
+    function completeTransition() {
+        flushCallbacks(gd._transitionData._interruptCallbacks);
+
+        return Promise.resolve().then(function() {
+            if(transitionConfig.redraw) {
+                return Plotly.redraw(gd);
+            }
+        }).then(function() {
+            gd.emit('plotly_endtransition', []);
+        });
+    }
+
+    function interruptPreviousTransitions() {
+        gd.emit('plotly_interrupttransition', []);
+
+        return executeCallbacks(gd._transitionData._interruptCallbacks);
+    }
+
+    for(i = 0; i < traceIndices.length; i++) {
+        traceIdx = traceIndices[i];
+        var contFull = gd._fullData[traceIdx];
+        var module = contFull._module;
+
+        if(!module) continue;
+
+        if(!module.animatable) {
+            var thisUpdate = {};
+
+            for(var ai in data[i]) {
+                thisUpdate[ai] = [data[i][ai]];
+            }
+
+            /*restyleList.push((function(md, data, traces) {
+                return function() {
+                    return Plotly.restyle(gd, data, traces);
+                };
+            }(module, thisUpdate, [traceIdx])));*/
+        }
+    }
+
+    var seq = [Plots.previousPromises, interruptPreviousTransitions, prepareTransitions, executeTransitions];
+    //seq = seq.concat(restyleList);
+
+    var plotDone = Lib.syncOrAsync(seq, gd);
+
+    if(!plotDone || !plotDone.then) plotDone = Promise.resolve();
+
+    return plotDone.then(function() {
+        gd.emit('plotly_begintransition', []);
+        return gd;
+    });
+};
+
+/**
+ * Animate to a keyframe
+ *
+ * @param {string} name
+ *      name of the keyframe to create
+ * @param {object} transitionConfig
+ *      configuration for transition
+ */
+Plotly.animate = function(gd, frameName, transitionConfig) {
```

Comment:
@rreusser Would you be opposed to making `Plotly.animate` optionally take array of frame names?

So that,

```js
Plotly.plot(gd, mock.data, mock.layout);
Plotly.addFrames(gd, mock.frames);
Plotly.animate(gd, mock.frames.map((f) => f.name));
```

Or maybe even make `Plotly.animate` take look into objects for `name` properties so that

```js
Plotly.plot(gd, mock.data, mock.layout);
Plotly.addFrames(gd, mock.frames);
Plotly.animate(gd, mock.frames);
```

would work. Thoughts?